### PR TITLE
Add type definition file

### PR DESCRIPTION
**Pull request by WorldMaker**
Adds a Typescript type definition file. I placed it in the lib directory since that where the final build goes for node, though I realize that was a force add into a gitignored directory, it seemed like a better idea than complicating your build script.

(I also updated the build and test script to get it to work in PowerShell on my system, simplifying them to further take advantage of node auto-adding the `./node_modules/.bin/` to the path when running package.json scripts.)

(See #1)

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    lastPoint(): PointLower;
+
+    areas(): number[];
+    area(index: number): number;
+    totalArea(): number;
+    perimeter(index: number): number;
+    perimeters(): number[];
+    totalPerimeter(): number;
+    reverse(): this;
+    tresholdArea(minArea: number): void; // spelling included
```
Comment (casperlamboo) [inline lib/index.d.ts]: What do you mean by 'spelling included'?

Comment (WorldMaker) [inline lib/index.d.ts]: That `tresholdArea` looks like a typo to me (shouldn't it be `thresholdArea`?) but I was making sure the type definition was accurate to the actual shape of the class. Alternately, I could fix the spelling in both places.

Comment (casperlamboo) [inline lib/index.d.ts]: yes, you are right. fixed typo in https://github.com/Doodle3D/clipper-js/commit/c19c9213e783c64da334d43182021ffecd8f82cf

**Comment by casperlamboo** [comment]
While you were working on this pr I pushed one commit changing the api a little bit (https://github.com/Doodle3D/clipper-js/commit/6884885efc10a2eab1747d265077598408fb1cb1). Sorry for this. Both `pointInShape` and `pointInPath` now have two optional arguments, `capitalConversion` and `integerConversion` both are booleans.

**Comment by WorldMaker** [comment]
No problem. Consistency is a good thing.

**Pull request merged**